cfy_manager/components/stage/stage.py

```diff
@@ -217,3 +217,8 @@ def remove(self):
         logger.notice('Removing Stage data....')
         common.sudo(['rm', '-rf', '/opt/cloudify-stage'])
         logger.notice('Stage successfully removed')
+
+    def upgrade(self):
+        logger.notice('Upgrading Cloudify Stage...')
+        self._run_db_migrate()
```

Comment:
A general comment about th _run_db_migrate() function. As in the prometheus component case,[ this function uses the key `CLUSTER_JOIN`](https://github.com/cloudify-cosmo/cloudify-manager-install/blob/master/cfy_manager/components/stage/stage.py#L75) which we already know is always `False`.
This means that on a 3 nodes cluster, the db migration will run 3 times on each node. Thi is true also for the composer component.
What do you think?

Reply:
`CLUSTER_JOIN` is not always `False`.  It's being read from `config.yaml` where sometimes (e.g. in case of two out of 3-nodes) it is actually `True`.

`_run_db_migrate()` shouldn't be called more than once in the 3-cluster environment.  This is because we'd be using three different files (`cfy_manager upgrade -c /etc/cloudify/{db,rabbit,manager}_config.yaml`).  It is only `manager_config.yaml` that has `manager_service` listed among `services_to_install`.  So Stage/Composer should be upgraded only once

Reply:
Right, it should have the `CLUSTER_JOIN` set to True, only that this is set in the restservice component, which we don't use in the upgrade process - https://github.com/cloudify-cosmo/cloudify-manager-install/blob/master/cfy_manager/components/restservice/restservice.py#L319.
This is why we added the condition here https://github.com/cloudify-cosmo/cloudify-manager-install/blob/11950b1ffed55fab4fba203288ea69272c353312/cfy_manager/components/prometheus/prometheus.py#L507. 
So if I understand correctly, we should add this condition for Stage and Composer too (unless I'm missing something).